openwisp_notifications/base/models.py

```diff
@@ -86,6 +120,33 @@ def email_subject(self):
         else:
             return self.message
 
+    def _related_object(self, field):
+        obj_id = getattr(self, f'{field}_object_id')
+        if not obj_id:
+            return
+        cache_key = self._cache_key(obj_id)
```

Comment:
please take in consideration the django content type as well when calculating the cache key, you could try to use the content type id to avoid generating the query to fetch the content type object